packages/google_sign_in/google_sign_in/lib/google_sign_in.dart

```diff
@@ -189,7 +189,13 @@ class GoogleSignIn {
     this.clientId,
     this.serverClientId,
     this.forceCodeForRefreshToken = false,
-  });
+  }) {
+    // Start initializing.
+    // Async methods in the plugin will await for this to be done.
+    if (kIsWeb) {
```

Comment:
Can we document why this is web-only?

Reply:
This doesn't really need to be web-only, but I didn't want to break other platforms by being slightly more eager when kickstarting the plugin initialization procedure.

Reply:
I've documented this bit of the code better, though!